src/Spatial/Euclidean/LIneSegment2D.cs

```diff
@@ -0,0 +1,224 @@
+﻿namespace MathNet.Spatial.Euclidean
+{
+    using System;
+    using System.Diagnostics.Contracts;
+    using MathNet.Spatial.Units;
+
+    /// <summary>
+    /// This structure represents a line between two points in 2-space.  It allows for operations such as
+    /// computing the length, direction, comparisons, and shifting by a vector.
+    /// </summary>
+    public struct LineSegment2D : IEquatable<LineSegment2D>
+    {
+        /// <summary>
+        /// The starting point of the line segment
+        /// </summary>
+        public readonly Point2D StartPoint;
+
+        /// <summary>
+        /// The end point of the line segment
+        /// </summary>
+        public readonly Point2D EndPoint;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LineSegment2D"/> struct.
+        /// Throws an ArgumentException if the <paramref name="startPoint"/> is equal to the <paramref name="endPoint"/>.
+        /// </summary>
+        /// <param name="startPoint">the starting point of the line segment.</param>
+        /// <param name="endPoint">the ending point of the line segment</param>
+        public LineSegment2D(Point2D startPoint, Point2D endPoint)
+        {
+            if (startPoint == endPoint)
+            {
+                throw new ArgumentException("The Line2D starting and ending points cannot be identical");
+            }
+
+            this.StartPoint = startPoint;
+            this.EndPoint = endPoint;
+        }
+
+        /// <summary>
+        /// Gets the distance from <see cref="StartPoint"/> to <see cref="EndPoint"/>
+        /// </summary>
+        [Pure]
+        public double Length => this.StartPoint.DistanceTo(this.EndPoint);
+
+        /// <summary>
+        /// Gets a normalized vector in the direction from <see cref="StartPoint"/> to <see cref="EndPoint"/>
+        /// </summary>
+        [Pure]
+        public Vector2D Direction => this.StartPoint.VectorTo(this.EndPoint).Normalize();
```

Comment:
You really hate `UnitVector3D` :) I think this should be `UnitVector3D` as long as it is around.

Reply:
there isn't a UnitVector2D ...

Reply:
oh, derp, my bad, sry